src/auth/ejabberd_auth.erl

```diff
@@ -328,6 +333,21 @@ does_stored_user_exist(HostType, #jid{luser = LUser, lserver = LServer}) ->
 does_stored_user_exist(_HostType, error) ->
     false.
 
+-spec does_user_exist(Status :: mongoose_hooks:simple_acc(),
+                      HostType :: mongooseim:host_type(),
+                      Jid :: jid:jid()) -> does_user_exist().
+does_user_exist(#{result := true} = Status, _, _) ->
```

Comment:
Instead of having this manual `result` check, we should just stop hook processing (return `{stop, NewAcc}` from the hook) if there's a cache record found

Reply:
Ok, we're going to think nobody's inserting a handler _after_ the cache has already said `true`, so I'm doing a stop here. This allows me to simplify the accumulator and the second and third handlers, as they won't even be called if the cache was already called.